crates/revm/src/db/states/state_builder.rs

```diff
@@ -29,26 +26,38 @@ pub struct StateBuilder<DB> {
     with_block_hashes: BTreeMap<u64, B256>,
 }
 
-impl Default for StateBuilder<Box<EmptyDB>> {
+impl StateBuilder<EmptyDB> {
+    /// Create a new builder with an empty database.
+    ///
+    /// If you want to instatiate it with a specific database, use
+    /// [`new_with_database`](Self::new_with_database).
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl<DB: Database + Default> Default for StateBuilder<DB> {
     fn default() -> Self {
+        Self::new_with_database(DB::default())
+    }
+}
+
+impl<DB: Database> StateBuilder<DB> {
+    /// Create a new builder with the given database.
+    pub fn new_with_database(database: DB) -> Self {
         Self {
+            database,
             with_state_clear: true,
-            database: Box::<EmptyDB>::default(),
             with_cache_prestate: None,
             with_bundle_prestate: None,
             with_bundle_update: false,
             with_background_transition_merge: false,
             with_block_hashes: BTreeMap::new(),
         }
     }
-}
-
-impl<DB: Database> StateBuilder<DB> {
-    /// Create default instance of builder.
-    pub fn new() -> StateBuilder<Box<EmptyDB>> {
-        StateBuilder::<Box<EmptyDB>>::default()
-    }
 
+    #[doc(hidden)]
+    #[deprecated = "instantiate with `new_with_database` instead"]
```

Comment:
I use this fn inside reth, maybe it is not needed as there is `fn with_database` currently in the main, but can you remove `deprecated` for now?